packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -131,3 +131,12 @@ export const createPluginActions: CreatePluginActions = (
     messageCallbacks,
   }
 }
+
+/**
+ * If we know that setStateAction is a SetStateAction, we can safely check the type with typeof setStateAction === 'function'.
+ * But typescript's does not manage to infer the type after such a check, thus we extract that logic into this utility function.
+ * @param setStateAction
+ */
+const isFunction = <T>(
+  setStateAction: SetStateAction<T>,
+): setStateAction is (state: T) => T => typeof setStateAction === 'function'
```

Comment:
If we would inline `typeof setStateAction === 'function'` above, Typescript wouldn't infer the type `(content: T) => T`, but instead infer the type `Function` -- even though we know it can only be one type of function.